### PR TITLE
Update naming of config keys, filenames, references and code to remove obsolete terms

### PR DESCRIPTION
**Pull request by merlante**
…nary, docker and generated files.

## Describe your changes

- Update refs of ciam-rebac to relations-api, incl packages, protos, binary, docker and generated files.

## Ticket reference (if applicable)
Fixes #[RHCLOUD-33338](https://issues.redhat.com/browse/RHCLOUD-33338)

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: api/health/v1/health.proto [hunk 1]
```diff
@@ -4,7 +4,7 @@ package kessel.relations.v1;
 
 import "google/api/annotations.proto";
 
-option go_package = "ciam-rebac/api/health/v1;v1";
+option go_package = "relations-api/api/health/v1;v1";
```
Comment (alechenninger) [inline api/health/v1/health.proto:7]: Does this go_package resolve properly?

See:

- https://stackoverflow.com/questions/61666805/correct-format-of-protoc-go-package
- https://protobuf.dev/reference/go/go-generated/#package

Comment (alechenninger) [reply api/health/v1/health.proto:7]: Proto docs also suggest we should drop the `;v1` part

File: go.mod [hunk 2]
```diff
@@ -1,4 +1,4 @@
-module ciam-rebac
+module relations-api
```
Comment (alechenninger) [inline go.mod:1]: Should this be the actual go import path? (e.g. github.com/project-kessel/relations-api ?) Or a domain?

This seems to be common in any case, so we might want to go with the github.com module name or like, project-kessel.org/relations-api or something like that

Comment (Rajagopalan-Ranganathan) [reply go.mod:1]: github.com/project-kessel/relations-api  or project-kessel.org/relations-api (repo vs domain)

Decision: use: github.com/project-kessel/relations-api

File: Dockerfile [hunk 3]
```diff
@@ -19,17 +19,17 @@ RUN make build
...
+COPY --from=builder /workspace/bin/relations-api /usr/local/bin/
 COPY --from=builder /workspace/configs/config.yaml /usr/local/bin/
 
 EXPOSE 8000
 EXPOSE 9000
 
 USER 1001
 
-ENTRYPOINT ["/usr/local/bin/ciam-rebac","-conf","/usr/local/bin/config.yaml"]
+ENTRYPOINT ["/usr/local/bin/relations-api","-conf","/usr/local/bin/config.yaml"]
```
Comment (alechenninger) [inline Dockerfile:30]: Should our binary have "kessel" in it? (or "ksl" if you want to keep it short)

e.g. kessel-relations or ksl-relations

File: api/health/v1/health.proto (hunk 1 above)
Comment (Rajagopalan-Ranganathan) [reply api/health/v1/health.proto:7]: 1) renaming package: to full import path (URL in the browser - after relationsa-pi - get the dir 
ex: github.com/project-kessel/relations-api/api/health/v1 

Make the package name and go_package the same

2) remove ;v1 and check  - deprecated as per: https://protobuf.dev/reference/go/go-generated/#package (if hugely disruptive - defer this - create a ticket for future)

File: Dockerfile (hunk 3 above)
Comment (Rajagopalan-Ranganathan) [reply Dockerfile:30]: name of binary: kessel-relations or ksl-relations

Decision: kessel-relations (for the server binary)

**Review by Rajagopalan-Ranganathan — APPROVED** [review]

**Pull request merged**